Remove Optional Argument from _fill_perfdata

michaelmckinsey1 opened the pull request:
Always fill numerical columns with `np.nan` to make functionality clearer.

ilumsden reviewed (approved):
Looks good.

One minor comment is that it's probably better to use `NaN` in the comments, but that's so minor that it's fine if you don't do that.

Also, whenever it's no longer experimental in pandas, we should probably switch to using `pandas.NA`. It's a pandas-specific type for representing missing data that works for all pandas data types. So, `NA` can be interpreted as `Int64Dtype`, `StringDtype`, `Float64Dtype`, `BooleanDtype`, and even the newer `ArrowDtype`.

@pearce8 this is ready for your review

michaelmckinsey1 committed:
Remove optional argument from _fill_perfdata to always fill with NaN

michaelmckinsey1 committed:
Fix spelling

michaelmckinsey1 commented:
> Looks good.
> 
> One minor comment is that it's probably better to use `NaN` in the comments, but that's so minor that it's fine if you don't do that.
> 
> Also, whenever it's no longer experimental in pandas, we should probably switch to using `pandas.NA`. It's a pandas-specific type for representing missing data that works for all pandas data types. So, `NA` can be interpreted as `Int64Dtype`, `StringDtype`, `Float64Dtype`, `BooleanDtype`, and even the newer `ArrowDtype`.
> 
> @pearce8 this is ready for your review

I reverted the change back to "NaN". They actually were "NaN" before and I had changed it to "nan" to match `np.nan`. However, since you mentioned it this doesn't make sense, as we should only be using `NaN` or `np.nan`, and `NaN` makes more sense in the context of comments.
